bcipy/io/load.py

```diff
@@ -318,3 +302,297 @@ def fast_scandir(directory_name: str, return_path: bool = True) -> List[str]:
         return [f.path for f in os.scandir(directory_name) if f.is_dir()]
 
     return [f.name for f in os.scandir(directory_name) if f.is_dir()]
+
+
+class BciPySessionTaskData:
+    """Session Task Data.
+
+    This class is used to represent a single task data session. It is used to store the
+    path to the task data, as well as the parameters and other information about the task.
+
+    /<local_path>/<user_id>/<date>/<experiment_id>/<date_time>/
+        protocol.json
+        <task_date_time>/
+            parameters.json
+            **task_data**
+
+    """
+
+    def __init__(
+            self,
+            path: str,
+            user_id: str,
+            date: str,
+            experiment_id: str,
+            date_time: str,
+            task: str,
+            run: int = 1) -> None:
+
+        self.user_id = user_id
+        self.date = date
+        self.experiment_id = experiment_id.replace('_', '')
+        self.date_time = date_time.replace("_", "").replace("-", "")
+        self.session_id = f'{self.experiment_id}{self.date_time}'
+        self.date_time = date_time
+        self.task = task
+        self.run = str(run)
+        self.path = path
+        self.parameters = self.get_parameters()
```

Comment:
You might consider whether parameters should always be loaded here. It makes it a heavy object to pass around if it's not being used. It might make sense to have parameters be a computed property that's only loaded when needed.
```
@property
def parameters(self):
    if not self._parameters:
        self._parameters = load_json_parameters(
            f'{self.path}/{DEFAULT_PARAMETERS_FILENAME}',
            value_cast=True)
    return self._parameters

Reply:
It only needs to grab the task name, but it can be used in the future to persist with some other relevant parameters when bundling. I will consider how to remove/standardize from the file structure.